packages/taiko-client/driver/preconf_blocks/server.go

```diff
@@ -0,0 +1,208 @@
+type PreconfBlockAPIServer struct {
+	echo               *echo.Echo
+	chainSyncer        preconfBlockChainSyncer
+	rpc                *rpc.Client
+	txListDecompressor *txListDecompressor.TxListDecompressor
+	checkSig           bool
+}
+
+// New creates a new preconf blcok server instance, and starts the server.
+func New(
+	cors string,
+	jwtSecret []byte,
+	chainSyncer preconfBlockChainSyncer,
+	cli *rpc.Client,
+	checkSig bool,
+) (*PreconfBlockAPIServer, error) {
+	protocolConfigs, err := cli.GetProtocolConfigs(nil)
+	if err != nil {
+		return nil, fmt.Errorf("failed to fetch protocol configs: %w", err)
+	}
+
+	server := &PreconfBlockAPIServer{
+		echo:        echo.New(),
+		chainSyncer: chainSyncer,
+		txListDecompressor: txListDecompressor.NewTxListDecompressor(
+			uint64(protocolConfigs.BlockMaxGasLimit()),
+			uint64(rpc.BlobBytes),
+			cli.L2.ChainID,
+		),
+		rpc:      cli,
+		checkSig: checkSig,
+	}
+
+	server.echo.HideBanner = true
+	server.configureMiddleware([]string{cors})
+	server.configureRoutes()
+	if jwtSecret != nil {
+		server.echo.Use(echojwt.JWT(jwtSecret))
+	}
+
+	return server, nil
+}
+
+// LogSkipper implements the `middleware.Skipper` interface.
+func LogSkipper(c echo.Context) bool {
+	switch c.Request().URL.Path {
+	case "/healthz":
+		return true
+	default:
+		return true
+	}
+}
+
+// configureMiddleware configures the server middlewares.
+func (s *PreconfBlockAPIServer) configureMiddleware(corsOrigins []string) {
+	s.echo.Use(middleware.RequestID())
+
+	s.echo.Use(middleware.LoggerWithConfig(middleware.LoggerConfig{
+		Skipper: LogSkipper,
+		Format: `{"time":"${time_rfc3339_nano}","level":"INFO","message":{"id":"${id}","remote_ip":"${remote_ip}",` +
+			`"host":"${host}","method":"${method}","uri":"${uri}","user_agent":"${user_agent}",` +
+			`"response_status":${status},"error":"${error}","latency":${latency},"latency_human":"${latency_human}",` +
+			`"bytes_in":${bytes_in},"bytes_out":${bytes_out}}}` + "\n",
+		Output: os.Stdout,
+	}))
+
+	// Add CORS middleware
+	s.echo.Use(middleware.CORSWithConfig(middleware.CORSConfig{
+		AllowOrigins:     corsOrigins,
+		AllowCredentials: true,
+	}))
+}
+
+// Start starts the HTTP server.
+func (s *PreconfBlockAPIServer) Start(port uint64) error {
+	return s.echo.Start(fmt.Sprintf(":%v", port))
+}
+
+// Shutdown shuts down the HTTP server.
+func (s *PreconfBlockAPIServer) Shutdown(ctx context.Context) error {
+	return s.echo.Shutdown(ctx)
+}
+
+// configureRoutes contains all routes which will be used by the HTTP server.
+func (s *PreconfBlockAPIServer) configureRoutes() {
+	s.echo.GET("/", s.HealthCheck)
+	s.echo.GET("/healthz", s.HealthCheck)
+	s.echo.POST("/preconfBlocks", s.BuildPreconfBlock)
+	s.echo.DELETE("/preconfBlocks", s.RemovePreconfBlocks)
+}
+
+// OnUnsafeL2Payload implements the p2p.GossipIn interface.
+func (s *PreconfBlockAPIServer) OnUnsafeL2Payload(
+	ctx context.Context,
+	from peer.ID,
+	msg *eth.ExecutionPayloadEnvelope,
```

Comment:
Why not use [taiko-geth/ExecutionPayloadEnvelope](https://github.com/taikoxyz/taiko-geth/blob/cdca79128bc606f89c12e08474f228ad5d0d89c3/beacon/engine/types.go#L138)?